storage/storage_dest.go

```diff
@@ -312,6 +312,7 @@ func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAcces
 	}
 
 	if out.TOCDigest == "" && out.UncompressedDigest == "" {
+		_ = s.imageRef.transport.store.CleanupStagedLayer(out)
```

Comment:
`storageImageDestination.Close` has this conditioned on `out.Target != nil`. Is that necessary here as well?

Quite possibly `storageImageDestination.Close` should just drop the condition instead.

Reply:
yes let's drop the condition from `storageImageDestination.Close` and let's c/storage deal with the cleanup